src/libraries/Microsoft.PowerFx.Interpreter/Functions/StandardErrorHandling.cs

```diff
@@ -128,6 +129,61 @@ private static bool IsInvalidDouble(double number)
             });
         }
 
+        #region Single Column Table Functions
+        public static Func<EvalVisitor, SymbolContext, IRContext, TableValue[], FormulaValue> StandardSingleColumnTable<T>(FormulaType itemType, Func<EvalVisitor, SymbolContext, IRContext, T[], FormulaValue> targetFunction) where T : FormulaValue
+        {
+            return (runner, symbolContext, irContext, args) =>
+            {
+                var tableType = (TableType)irContext.ResultType;
+                var resultType = tableType.ToRecord();
+
+                var arg0 = args[0];
+                var resultRows = new List<DValue<RecordValue>>();
+                foreach (var row in arg0.Rows)
+                {
+                    if (row.IsValue)
+                    {
+                        var value = row.Value.GetField(BuiltinFunction.ColumnName_ValueStr);
+                        NamedValue namedValue;
+                        if (value is T t)
```

Comment:
could this be cleaner as a pattern matching switch statement?

Reply:
Done